core/src/main/scala/org/apache/spark/scheduler/dynalloc/ExecutorMonitor.scala

```diff
@@ -279,5 +452,35 @@ private[spark] class ExecutorMonitor(
         updateNextTimeout(newDeadline)
       }
     }
+
+    def addShuffle(id: Int): Unit = {
+      if (shuffleIds.add(id)) {
+        hasActiveShuffle = true
+        if (!isIdle) {
+          updateTimeout()
```

Comment:
`isIdle` is always false here, because you've just set `hasActiveShuffle=true`, right?  I guess I am not sure why you'd need to call updateTimeout() here so I'm not sure the right condition to check ...

Reply:
I think this can go away; I changed `onTaskEnd` at some point so that the task count is updated after this call, so that code should handle the necessary updates.